e3sm_diags/driver/arm_diags_driver.py

```diff
@@ -23,179 +27,166 @@
+                                ds_ref, var, season, fft=False
                             )
-                            if hasattr(ref, "standard_name"):
-                                ref.long_name = ref.standard_name
 
                             ref = ref_diurnal
 
                 else:
-                    ref_data = utils.dataset.Dataset(parameter, ref=True)
-                    ref = ref_data.get_timeseries_variable(var, single_point=True)
-                    ref.lat = test_data.get_static_variable("lat", var)
-                    ref.lon = test_data.get_static_variable("lon", var)
-                    ref_diurnal, lst = utils.diurnal_cycle.composite_diurnal_cycle(
-                        ref, season, fft=False
+                    ref_data = Dataset(parameter, data_type="ref")
+                    ds_ref = ref_data.get_time_series_dataset(var, single_point=True)
+
+                    ref_diurnal, lst = composite_diurnal_cycle(  # type: ignore
+                        ds_ref, var, season, fft=False
                     )
                     ref = ref_diurnal
 
                 refs.append(ref)
 
+                # Create the metrics dictionary.
                 metrics_dict = {}
+                metrics_dict["unit"] = ds_test[var].units
+
+                # Update the result metrics dictionary and store it in the vars_
+                # to_data dictionary.
                 result = RefsTestMetrics(
                     test=test_diurnal, refs=refs, metrics=None, misc=lst
                 )
                 vars_to_data[season] = result
-                # Saving the metrics as a json.
-                metrics_dict["unit"] = test.units
+
                 parameter.output_file = "-".join([ref_name, var, season, region])
-                fnm = os.path.join(
-                    utils.general.get_output_dir(parameter.current_set, parameter),
-                    parameter.output_file + ".json",
-                )
-                with open(fnm, "w") as outfile:
-                    json.dump(metrics_dict, outfile)
-                # Get the filename that the user has passed in and display that.
-                fnm = os.path.join(
-                    utils.general.get_output_dir(parameter.current_set, parameter),
-                    parameter.output_file + ".json",
-                )
-                logger.info("Metrics saved in: " + fnm)
+                _save_metrics_to_json(parameter, metrics_dict)
 
-                arm_diags_plot.plot_diurnal_cycle(var, vars_to_data[season], parameter)
+                # Set the plot and viewer output attributes.
+                parameter.viewer_descr[var] = ds_test[var].attrs.get("long_name", var)
+                parameter.test_name_yrs = test_ds.get_name_yrs_attr()
+                parameter.var_name = ds_test[var].attrs.get("long_name", var)
+                parameter.var_units = ds_test[var].attrs.get("units", var)
```

Comment:
Integration test breaks for `CLOUD` variable because `long_name` attr does not exist.

I added `.get()` statements for attributes with default values. This replicates the CDAT codebase behavior.